homeassistant/components/broadlink/climate.py

```diff
@@ -0,0 +1,120 @@
+"""Support for Broadlink climate devices."""
+import logging
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import callback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Broadlink climate entities."""
+    device = hass.data[DOMAIN].devices[config_entry.entry_id]
+
+    if device.api.type in {"Hysen heating controller"}:
+        climate_entities = [BroadlinkHysen(device)]
+    else:
+        _LOGGER.warning("Unrecognised broadlink climate API.")
+    async_add_entities(climate_entities)
+
+
+class BroadlinkHysen(ClimateEntity):
+    """Representation of a Broadlink Hysen climate entity."""
+
+    def __init__(self, device):
+        """Initialize the climate entity."""
+        self._device = device
+        self._coordinator = device.update_manager.coordinator
+        self._supported_features = SUPPORT_TARGET_TEMPERATURE
+        self._name = "Broadlink Thermosthat"
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._supported_features
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        # self._coordinator.data["thermostat_temp"] = temperature // maybe useless
+        if temperature < self.current_temperature + 0.5:
+            self._coordinator.data["active"] = 0
+        else:
+            self.set_hvac_mode(HVAC_MODE_HEAT)
+        self._device.api.set_temp(temperature)
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 0.5
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._coordinator.data["room_temp"]
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement that is used."""
+        return TEMP_CELSIUS
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        return self._coordinator.data["thermostat_temp"]
+
+    @property
+    def hvac_mode(self):
+        """Return current operation ie. heat, cool, idle."""
+        if (
+            self._coordinator.data["power"]
+            and self.current_temperature + 0.5 > self.target_temperature
```

Comment:
We don't calculate state and this property should return what mode the user has set.

Reply:
I think we should split HVAC_MODE and HVAC_ACTION.

```python3
    @property
    def hvac_mode(self):
        return HVAC_MODE_HEAT if self._coordinator.data["power"] else HVAC_MODE_OFF

    @property
    def hvac_action(self):
        return CURRENT_HVAC_HEAT if self._coordinator.data["active"] else CURRENT_HVAC_OFF
```

Reply:
If we need to add more logic, we can do it inside update_data(), example:
```python3
    @callback
    def update_data(self):
        """Update data."""
        self._hvac_mode = HVAC_MODE_HEAT if self._coordinator.data["power"] else HVAC_MODE_OFF
        self._hvac_action = CURRENT_HVAC_HEAT if self._coordinator.data["active"] else CURRENT_HVAC_OFF
        self.async_write_ha_state()
```